src/object.d

```diff
@@ -1718,6 +1745,21 @@ class Throwable : Object
      */
     @system @nogc final pure nothrow ref uint refcount() return scope { return _refcount; }
 
+    /**
+     * Loop over the chain of Throwables.
+     */
+    int opApply(scope int delegate(ref Throwable) dg)
+    {
+        int result = 0;
+        for (Throwable t = this; t; t = t.nextInChain)
+        {
+            result = dg(t);
+            if (result)
+                break;
+        }
+        return result;
```

Comment:
code is not covered

Reply:
It's tested here: https://github.com/dlang/dmd/pull/7832

Reply:
Ranges don't work in druntime because they require things that are in Phobos. Ranges are a Phobos feature. opApply was used here to hide access to the list from users.

You're right that the delegate should be scope, but due to a bug in the compiler that doesn't work. I haven't gotten around to fixing that bug yet, but that shouldn't block this PR. The scope can always be added later.

Reply:
> Ranges don't work in druntime because they require things that are in Phobos. 

I don't follow. Sure, lot's of goodies are in `std.range`, but what stops your from just defining `popFront`, `front`, `save` and `empty`?

Reply:
You're right the `ref` was unnecessary. Removed it.

Reply:
We really should use the range primitives here. Let me sketch the implementation below - it's just a few lines of code:

```d
/// Provides forward iteration for the chain positioned on this `Throwable` object
auto opSlice()
{
    static struct Result
    {
        private Throwable current;
        bool empty() { return current is null; }
        Throwable front() { return current; }
        void popFront() { assert(current); current = current.nextInChain; }
        Result save() { return this; }
    }
    return Result(this);
}
```

With this we get to avoid a variety of issues that `opApply` has and make client code a whole lot more flexible.

Reply:
The point of encapsulating the loop is I want to *completely hide* the linked list from the user, not merely provide a generic way of iterating it. This is because I want to prevent instances escaping without going through the ref counting mechanism. opApply is a better tool for that.

Reply:
@WalterBright I understand. We could do it in the range by incrementing the ref counts in `this(this)` and `save()`, but granted that complicates matters. I'll approve.